src/comm/aca_comm_mgr.cpp

```diff
@@ -69,6 +69,37 @@ int Aca_Comm_Manager::deserialize(const cppkafka::Buffer *kafka_buffer, GoalStat
   }
 }
 
+int Aca_Comm_Manager::deserialize(const void *pulsar_buffer, size_t buffer_length, GoalState &parsed_struct)
```

Comment:
both the deserialize for pulsar_buffer and deserialize for kafka_buffer function does very simlar things. A better way to deal with it is to make a generic deserialize function to handle both:
int Aca_Comm_Manager::deserialize(const void *MQ_buffer, size_t buffer_length, GoalState &parsed_struct)

And then update the call from both kafka path and pulsar path to provide the correct information to the new function above.